File: src/test/java/org/dedda/bratwurst/parse/InstanceofParserTest.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.BratwurtstTestcase;
import org.dedda.bratwurst.lang.Instanceof;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

/**
 * Created by dedda on 12/9/15.
 *
 * @author dedda
 */
public class InstanceofParserTest extends BratwurtstTestcase {

    @Test
    public void testParse() throws Exception {
        String line = "testVar -?> testClass";
        InstanceofParser parser = new InstanceofParser();
        Instanceof instruction = parser.parse(line, 0);
        assertEquals("testClass", instruction.className);
    }
}

File: src/test/java/org/dedda/bratwurst/parse/InstanceofParserTest.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.BratwurtstTestcase
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 12/9/15.
 *
 * @author dedda
 */
class InstanceofParserTest : BratwurtstTestcase() {
    @Test
    @Throws(Exception::class)
    fun testParse() {
        val line = "testVar -?> testClass"
        val parser = InstanceofParser()
        val instruction = parser.parse(line, 0)
        Assert.assertEquals("testClass", instruction.className)
    }
}